### DISCUSSION TITLE
One secret many devices.

### DISCUSSION THREAD

**Discussion by remchuk** [Onemind-Services-LLC/netbox-secrets#87] [Q&A]
Is there a way to add the same secret to many devices? We want the device to be depended on the secret and not the other way.

**Comment by jhofmueller** [comment]
I have a similar problem. Assigning scretes to multiple objects will not work according to a comment in `netbox_secrets/models/secrets.py`. So our thought was we assign a scret to a `device-role` to achieve this.
Adding `dcim.device_role` to the `apps` list did not produce the expected result (a box named `secrets` on the device-role page). Any thoughts on that?

**Comment by jhofmueller** [comment]
I solved the problem by finding the correct way to configure the `netbox_secrets` plugin. If you add `dcim.devicerole` to the configuration you can add secrets to device roles and thus use the same secret on many devices. This is what my `plugins.py` looks like:

```
PLUGINS = [
'netbox_secrets',
]

PLUGINS_CONFIG = {
'netbox_secrets': {'apps': ['dcim.device', 'dcim.devicerole'], 'enable_contacts': True},
}
```
